### DISCUSSION TITLE
Add a way to hoist meta and link tags to <head>

### DISCUSSION THREAD

**Discussion by RafidMuhymin** [withastro/roadmap#56] [Proposal]
There should be a way to hoist `<meta>` and `<link>` (or any other tag that should be inside `<head>`) to the `<head>` of the page. IIRC it was done by default before `v0.21`. And it's a necessary feature also I think.

An example use case:

Most of the images on a webpage should be lazily loaded. But some images like banners, hero images, or logos (it should be inline though) should be *eagerly* loaded. To do this, one should not only tell the browser to eagerly load the image using `loading="eager"` but he should also preload the image. For this, users can pass a format using the `preload` prop to the `<Image>` component to *eagerly preload* the image.

But when I was implementing this feature I found that unlike `v0.20`, `<link>`, `<meta>` or other tags that should be in the `<head>` of the page aren't hoisted.

So, not only the `<Image>` component is affected but any third-party package or a user who wants to implement a similar thing/feature can't do this.

**Comment by boehs** [comment]
1. Head should always be hoisted, I support this
2. My website is unorthodox in style. Instead of setting head stuff manually for each page I do it in my layout

```html
<html lang="en">
  <head>
    {noindex ? <meta name="robots" content="noindex"> : null}
    {warn ? <meta name="rating" content="adult"> : null}
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    <meta charset="utf-8">
    <meta name="theme-color" content="#0aa06e"/>
    <meta name="author" content={config.author}>
    <meta name="description" content={Astro.props.is ? `${config.author} is ${Astro.props.is}.` : `Some ramblings about ${Astro.props.title}!`}>
    <title>{title}</title>
  </head>
  <body>
    <div>
      <article>
        <slot name="title">
          <h2>{title}</h2>
        </slot>
        <slot/>
      </article>
      <Footer/>
    </div>
```

**Comment by tony-sull** [comment]
There's an accepted RFC to finalize `<html>`, `<head>`, and `<body>` behavior.  Take a look and see if it already covers the use cases you have, if not feel free to add a comment on there with use cases you'd like to see considered!

https://github.com/withastro/rfcs/blob/main/active-rfcs/0007-finalize-head-body.md
